<a href="https://colab.research.google.com/github/PradyumnaGupta/FLIPR_HACKATHON_TASK/blob/master/FLipr_TASK_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Diuresis Prediction on 27th MARCH

Importing Libraries

In [0]:
import pandas as pd
import numpy as np
from sklearn.metrics import mean_squared_error
from math import sqrt

Importing Data

In [0]:
train=pd.read_csv('/content/Train_diuresis_dataset.csv')
train

people_ID  20-Mar-20  21-Mar-20  ...  24-Mar-20  25-Mar-20 26-Mar-20
0              1        441        544  ...     1,002      1,216     1,473 
1              2        151        197  ...       400        493       607 
2              3        416        515  ...       948      1,150     1,396 
3              4        410        506  ...       928      1,127     1,368 
4              5        390        483  ...       893      1,084     1,316 
...          ...        ...        ...  ...        ...        ...       ...
10709      22691        134        174  ...       359        444       549 
10710      22692        387        479  ...       885      1,075     1,303 
10711      22693        177        228  ...       448        551       677 
10712      22694        352        438  ...       815        991     1,204 
10713      22695        181        230  ...       455        559       687 

[10714 rows x 8 columns]

Processing train values which are strings like "1,002" to "1002"

In [0]:
for i in range(train.shape[0]):
  for j in range(train.shape[1]):
    if type(train.iloc[i,j])==str:
      train.iloc[i,j]=train.iloc[i,j].replace(',','')

Preparing DataFrame for 27th March Values 

In [0]:
a=np.ones([train.shape[0],1])*np.nan
X_train=pd.concat([train.drop(['people_ID'],axis=1).astype(np.float64).iloc[:,:],pd.DataFrame(a)],axis=1)
columns=list(X_train.columns[:-1])
columns.append('27-Mar-20')
X_train.columns=columns
X_train

20-Mar-20  21-Mar-20  22-Mar-20  ...  25-Mar-20  26-Mar-20  27-Mar-20
0          441.0      544.0      670.0  ...     1216.0     1473.0        NaN
1          151.0      197.0      253.0  ...      493.0      607.0        NaN
2          416.0      515.0      632.0  ...     1150.0     1396.0        NaN
3          410.0      506.0      621.0  ...     1127.0     1368.0        NaN
4          390.0      483.0      595.0  ...     1084.0     1316.0        NaN
...          ...        ...        ...  ...        ...        ...        ...
10709      134.0      174.0      225.0  ...      444.0      549.0        NaN
10710      387.0      479.0      592.0  ...     1075.0     1303.0        NaN
10711      177.0      228.0      288.0  ...      551.0      677.0        NaN
10712      352.0      438.0      543.0  ...      991.0     1204.0        NaN
10713      181.0      230.0      290.0  ...      559.0      687.0        NaN

[10714 rows x 8 columns]

Predicting 27th March Diuresis values using Cubic SPline Interpolation

In [0]:
X_train.columns=range(X_train.shape[1])

predict=X_train.transpose().interpolate(method='spline',order=3,axis=0) #CUBIC SPLINE INTERPOLATION FOR TIME SERIES

predict=predict.transpose()
predict.columns=columns
predict

20-Mar-20  21-Mar-20  22-Mar-20  ...  25-Mar-20  26-Mar-20    27-Mar-20
0          441.0      544.0      670.0  ...     1216.0     1473.0  1772.142857
1          151.0      197.0      253.0  ...      493.0      607.0   737.714286
2          416.0      515.0      632.0  ...     1150.0     1396.0  1683.907344
3          410.0      506.0      621.0  ...     1127.0     1368.0  1650.000000
4          390.0      483.0      595.0  ...     1084.0     1316.0  1586.285714
...          ...        ...        ...  ...        ...        ...          ...
10709      134.0      174.0      225.0  ...      444.0      549.0   669.925442
10710      387.0      479.0      592.0  ...     1075.0     1303.0  1569.571429
10711      177.0      228.0      288.0  ...      551.0      677.0   824.285714
10712      352.0      438.0      543.0  ...      991.0     1204.0  1453.428571
10713      181.0      230.0      290.0  ...      559.0      687.0   833.428571

[10714 rows x 8 columns]